src/utils/chart-utils.tsx

```diff
@@ -0,0 +1,166 @@
+  },
+  { name: strings.timebank.expected, expected: personTotalTime.expected }
+];
+
+/**
+ * Renders custom labels in the pie chart
+ *
+ * @param props Props from the pie chart data, such as name and value
+ * @returns string
+ */
+export const renderCustomizedLabel = (props: CustomLabel) =>
+  `${props.name} ${getHoursAndMinutes(props.value)}`;
+
+/**
+ * Renders a customized tooltip when hovering over the chart. Pie chart version.
+ *
+ * @param props props, such as displayed data (payload), passed from the parent (chart)
+ * @returns JSX element as a tooltip
+ */
+export const renderCustomizedTooltipPieChart = (props: TooltipProps<ValueType, NameType>) => {
+  const { active, payload } = props;
+  if (!active || !payload || !payload.length) {
+    return null;
+  }
+
+  const selectedData = payload[0];
+
+  if (!selectedData.value || !selectedData.name) {
+    return null;
+  }
+
+  const sectionName = {
+    [strings.timebank.billableProject]: strings.timebank.billableProject,
+    [strings.timebank.nonBillableProject]: strings.timebank.nonBillableProject,
+    [strings.timebank.internal]: strings.timebank.internal
+  }[selectedData.name];
+
+  return (
+    <Box style={{ backgroundColor: "rgba(0, 0, 0)", opacity: "70%" }}>
+      <Typography
+        variant="h6"
+        style={{
+          color: "#fff",
+          padding: theme.spacing(1)
+        }}
+      >
+        {`${sectionName}: ${getHoursAndMinutes(selectedData.value as number)}`}
+      </Typography>
+    </Box>
+  );
+};
+
+/**
+ * Renders a customized tooltip when hovering over the chart. Bar chart version.
+ *
+ * @param props props, such as chart values, passed from the parent (chart)
+ * @returns JSX element as a tooltip
+ */
+export const renderCustomizedTooltipBarChart = (props: TooltipProps<ValueType, NameType>) => {
+  const { active, payload } = props;
+
+  if (!active || !payload || !payload.length || !payload[0].payload) {
+    return null;
+  }
+
+  const { billableProject, nonBillableProject, internal, expected, name } = payload[0].payload;
+
+  /**
+   * Renders MUI typography element inside the tooltip
+   *
+   * @param name Data name
+   * @param time Data value, time in minutes
+   * @param color Font color of the line
+   * @returns MUI Typography line inside the tooltip containing data name and value.
+   */
+  const renderCustomizedTooltipRow = (name: string, time: number, color: string) => (
+    <Typography
+      variant="h6"
+      style={{
+        color: color,
+        padding: theme.spacing(1)
+      }}
+    >
+      {`${name}: ${getHoursAndMinutes(time)}`}
+    </Typography>
+  );
+
+  return (
+    <Box
+      style={{
+        padding: theme.spacing(1),
+        backgroundColor: "#fff",
+        border: "1px solid rgba(0, 0, 0, 0.4)"
+      }}
+    >
+      <Typography
+        variant="h6"
+        style={{
+          padding: theme.spacing(1)
+        }}
+      >
+        {name}
+      </Typography>
+      {billableProject &&
```

Comment:
Why are all these conditionally rendered?

Reply:
Correction, conditional rendering does matter after all! This actually affects the overview (top chart) labels, without conditions you get a bunch of NaN values on the screen. Therefore, bringing conditions back.